app/src/main/java/com/example/scanpal/EventDetailsFragment.java

```diff
@@ -0,0 +1,160 @@
+package com.example.scanpal;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.Collection;
+
+public class EventDetailsFragment extends Fragment {
+
+    private String eventName;
+    private String eventDescription;
+    private String eventOrganizer;
+    private String eventLocation;
+    //private Collection<DocumentReference> eventAttendees; not really needed
+
+    //future field here for profile picture
+    //future field here for event poster/banner
+
+    /**
+     * Empty Constructor
+     */
+    public EventDetailsFragment() {
+
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.event_details, null, false);
+
+        //retrieves all the info about specific event from database
+        fetchEventDetails();
+
+        Log.d("fire", "end of onCreate");
+
+        return view;
+    }
+
```

Comment:
you might want to add navigation for the back button to go back to the events page

Reply:
Got it!